core/blockchain.go

```diff
@@ -562,110 +512,41 @@ func (bc *BlockChain) warmAcceptedCaches() {
 }
 
-// startAcceptor starts processing items on the [acceptorQueue]. If a [nil]
-// object is placed on the [acceptorQueue], the [startAcceptor] will exit.
-func (bc *BlockChain) startAcceptor() {
-	log.Info("Starting Acceptor", "queue length", bc.cacheConfig.AcceptorQueueLimit)
-
-	for next := range bc.acceptorQueue {
-		start := time.Now()
-		acceptorQueueGauge.Dec(1)
-
-		if err := bc.flattenSnapshot(func() error {
-			return bc.stateManager.AcceptTrie(next)
-		}, next.Hash()); err != nil {
-			log.Crit("unable to flatten snapshot from acceptor", "blockHash", next.Hash(), "err", err)
-		}
-
-		// Update last processed and transaction lookup index
-		if err := bc.writeBlockAcceptedIndices(next); err != nil {
-			log.Crit("failed to write accepted block effects", "err", err)
-		}
-
-		// Ensure [hc.acceptedNumberCache] and [acceptedLogsCache] have latest content
-		bc.hc.acceptedNumberCache.Put(next.NumberU64(), next.Header())
-		logs := bc.collectUnflattenedLogs(next, false)
-		bc.acceptedLogsCache.Put(next.Hash(), logs)
-
-		// Update the acceptor tip before sending events to ensure that any client acting based off of
-		// the events observes the updated acceptorTip on subsequent requests
-		bc.acceptorTipLock.Lock()
-		bc.acceptorTip = next
-		bc.acceptorTipLock.Unlock()
-
-		// Update accepted feeds
-		flattenedLogs := types.FlattenLogs(logs)
-		bc.chainAcceptedFeed.Send(ChainEvent{Block: next, Hash: next.Hash(), Logs: flattenedLogs})
-		if len(flattenedLogs) > 0 {
-			bc.logsAcceptedFeed.Send(flattenedLogs)
-		}
-		if len(next.Transactions()) != 0 {
-			bc.txAcceptedFeed.Send(NewTxsEvent{next.Transactions()})
-		}
-
-		bc.acceptorWg.Done()
+// accept processes a block that has been verified and updates the snapshot
+// and indexes.
+func (bc *BlockChain) accept(next *types.Block) error {
+	start := time.Now()
 
-		acceptorWorkTimer.Inc(time.Since(start).Milliseconds())
-		acceptorWorkCount.Inc(1)
-		// Note: in contrast to most accepted metrics, we increment the accepted log metrics in the acceptor queue because
-		// the logs are already processed in the acceptor queue.
-		acceptedLogsCounter.Inc(int64(len(logs)))
+	if err := bc.flattenSnapshot(func() error {
+		return bc.stateManager.AcceptTrie(next)
+	}, next.Hash()); err != nil {
+		log.Crit("unable to flatten snapshot from acceptor", "blockHash", next.Hash(), "err", err)
 	}
-}
-
-// addAcceptorQueue adds a new *types.Block to the [acceptorQueue]. This will
-// block if there are [AcceptorQueueLimit] items in [acceptorQueue].
-func (bc *BlockChain) addAcceptorQueue(b *types.Block) {
-	// We only acquire a read lock here because it is ok to add items to the
-	// [acceptorQueue] concurrently.
-	bc.acceptorClosingLock.RLock()
-	defer bc.acceptorClosingLock.RUnlock()
 
-	if bc.acceptorClosed {
-		return
+	// Update last processed and transaction lookup index
+	if err := bc.writeBlockAcceptedIndices(next); err != nil {
+		log.Crit("failed to write accepted block effects", "err", err)
```

Comment:
should we return errs here instead of just log them? I think we should keep the logs though.

Reply:
log.Crit uses `os.Exit` so it seems it should be fine, no?